### DISCUSSION TITLE
Access toolbar button from File menu

### DISCUSSION THREAD

**Discussion by imed-ghomari** [chrisgurney/obsidian-note-toolbar#137] [Ideas]
Hi, thank you for the plugin and its continuous development.

I was thinking of a quick and easy way to integrate your plugin with Dataview queries. I thought that we could just right-click the file link, and the toolbar's buttons would appear in the right-click menu, provided that this file has a toolbar attributed to it.

Would this be feasible?

This is related to #110 and #64, but not quite the same, correct me if I'm wrong.

**Comment by chrisgurney** [comment]
@imeed166 Thanks for your idea! 

Can you provide an example about how this ties in with Dataview specifically? Where exactly are you envisioning the right-click menu is being triggered from? (And are you suggesting the toolbar is added to an existing right-click menu?)

**Reply by imed-ghomari** [reply]
Yes, of course, sorry for not explaining it better.

So, as shown in the image, I right-click the link of the file queried by the Dataview query, which triggers the official Obsidian right-click menu. I envision the buttons appearing in their own section in this menu because the file queried has a toolbar attributed to it.

![Screen Shot 2024-09-14 at 12 46 29 PM](https://github.com/user-attachments/assets/c3f06e0d-83d7-4ad5-99e5-697817acc7e1)

**Reply by chrisgurney** [reply]
Thank you, that helps! I'm curious, what tools would you use to operate on a note that you haven't yet opened?

**Reply by imed-ghomari** [reply]
Here's few examples:

- I have some recurring tasks, so I would change the due date to the next recurrence using a Templater script linked to a command triggered by your button.
- I would complete some tasks/projects by changing the property (completed = true), using the same method described above.
- I would postpone tasks by changing the date for waiting tasks, using the same method.
- I would delete obsolete tasks/projects directly, as Obsidian’s default menu doesn’t provide this functionality, using the same method.

**Reply by chrisgurney** [reply]
@imeed166 Nice! I like these ideas.

I think I _might_ be able to accomplish this. Just need to figure out how to get the toolbar for a file that's _not_ the current file. i.e., I think I have to load the file to get its properties so I can double-check them for a toolbar mapping (in addition to its folder mapping).

I'll dig into this, for sure. Thanks!

**Reply by chrisgurney** [reply]
@imeed166 Another thought is whether users might want to use the current note's toolbar in the right-click menu, versus the linked note's toolbar.

I'm not sure I have a use case in mind, but something to consider is if I have to perhaps display two sub-menus: one with items for the current note's toolbar, and one with the linked note's toolbar. 

(If they're the same, then maybe they can be combined, but there's the question of what to do about resolving [variables](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Variables).)

**Reply by imed-ghomari** [reply]
If I understood correctly, you mean the toolbar for the note containing the query and the toolbar for the note queried, is that it ?

If yes, then I would definitely consider not displaying the current note's toolbar in the right-click menu, as it is already easily accessible. In my opinion, only display the toolbar of the linked note in the right click menu.

**Reply by chrisgurney** [reply]
@imeed166 Good thought. I'm learning towards agreeing with you here.

I have a prototype of this working locally, and I've chosen to put all of the toolbar's items under the toolbar's name, as a sub-menu.

I still need to work out the placement of sub-menus _within_ those sub-menus, however. If this proves to be an issue I can't solve, I _may_ hide sub-menus. 

Thanks again for the idea!

<img src="https://github.com/user-attachments/assets/8d650aaa-6b67-4a6b-8296-f27920214706" width="600"/>

**Comment by chrisgurney** [comment]
Now available (and hopefully working for you) in [release 1.13.3](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.13.3).

**Reply by imed-ghomari** [reply]
yes, it worked, thank you.
